std_execution.bs

```diff
@@ -553,6 +553,293 @@ merged into a single one in an asynchronous fashion.
 `transfer_when_all` accepts an additional scheduler argument. It returns a sender whose value [=completion scheduler=] is the scheduler provided as an argument, but otherwise behaves the same as `when_all`. You can think of it as a composition of
 `transfer(when_all(inputs...), scheduler)`, but one that allows for better efficiency through customization.
 
+## Senders support cancellation ## {#design-cancellation}
+
+Senders are often used in scenarios where the application may be concurrently executing
+multiple strategies for achieving some program goal. When one of these strategies succeeds
+(or fails) it may not make sense to continue pursuing the other strategies as their results
+are no longer useful.
+
+For example, we may want to try to simultaneously connect to multiple network servers and use
+whichever server responds first. Once the first server responds we no longer need to continue
+trying to connect to the other servers.
+
+Ideally, in these scenarios, we would somehow be able to request that those other strategies
+stop executing promptly so that their resources (e.g. cpu, memory, I/O bandwidth) can be
+released and used for other work.
+
+While the design of senders has support for cancelling an operation before it starts
+by simply destroying the sender or the operation-state returned from `execution::connect()`
+before calling `execution::start()`, there also needs to be a standard, generic mechanism
+to ask for an already-started operation to complete early.
+
+The ability to be able to cancel in-flight operations is fundamental to supporting some kinds
+of generic concurrency algorithms.
+
+For example:
+* a generic `timeout(src, duration)` algorithm needs to be able to cancel the `src` operation after the timeout duration has elapsed.
+* a `first_successful(ops...)` algorithm should cancel the other operations as soon as one operation completes successfuly
+* a `when_all(ops...)` algorithm should cancel other operations as soon as one operation fails
+* a `stop_when(src, trigger)` algorithm should cancel `src` if `trigger` completes first and cancel `trigger` if `src` completes first
```

Comment:
See the comment below; I feel that suddenly introducing a number of algorithms that exist solely for the purpose of explaining why another piece of machinery needs to exist goes contrary to what the rest of the paper has tried to do; I'm especially troubled by the fact that `when_all` is buried in the middle of this list.

Reply:
These algorithms are not solely for the purpose of explaining why cancellation exists - the intent is to provide algorithms like these in the standard library. That they are not included in this paper is mainly a scoping exercise.

There are few algorithms in this paper that actually have an interaction with cancellation - `when_all()` being the only algorithm that actually introduces a new cancellation-scope. There are a large number of other algorithms not yet included in this paper that do provide more facilities for dealing with cancellation.

Reply:
Put `when_all` first :)